podcasts/DiscoverViewController.swift

```diff
@@ -242,7 +242,12 @@ class DiscoverViewController: PCViewController {
         currentSnapshot = snapshot
     }
 
-    private func populateFrom(discoverLayout: DiscoverLayout?) {
+    /// Populate the scroll view using a DiscoverLayout with optional shouldInclude and shouldReset filters.
+    /// - Parameters:
+    ///   - discoverLayout: A `DiscoverLayout`
+    ///   - shouldInclude: Whether a `DiscoverItem` from the layout should be included in the scroll view. This is used to filter items meant only for certain categories, for instance.
+    ///   - shouldReset: Whether a view controller from `summaryViewControllers` should be reset during this operation. This is used by the Categories pills to avoid triggering a view reload, allowing animations to continue.
+    private func populateFrom(discoverLayout: DiscoverLayout?, shouldInclude: ((DiscoverItem) -> Bool)? = nil, shouldReset: ((DiscoverItem) -> Bool)? = nil) {
```

Comment:
I don't love the implementation of this but I think it needs to change to a `UICollectionView` or something sooner rather than later. I started looking at that and have made some good progress so will try to submit that some time next week, maybe.